File: backend/internal/infrastructure/persistence/mysql/model/message.go
package model

type Message struct {
	MessageId      string  `gorm:"size:32;primaryKey" json:"messageId"`
	ConversationId string  `gorm:"size:64;not null;index:idx_conv_seq,priority:1;uniqueIndex:uk_message_conv_seq,priority:1" json:"conversationId"`
	SendId         string  `gorm:"size:32;not null;index:idx_sender;uniqueIndex:uq_sender_client_msg,priority:1" json:"sendId"`
	ClientMsgId    *string `gorm:"size:64;uniqueIndex:uq_sender_client_msg,priority:2" json:"clientMsgId,omitempty"`
	RequestHash    string  `gorm:"size:64;default:'';index" json:"-"`

	Seq     int64  `gorm:"not null;index:idx_conv_seq,priority:2,sort:desc;uniqueIndex:uk_message_conv_seq,priority:2" json:"seq"`
	Type    int8   `gorm:"not null;comment:1=text 2=image 3=video 4=sticker 5=file" json:"type"`
	Content string `gorm:"type:text" json:"content"`
	// 弹幕/视频时间戳关联（非媒体字段）
	VideoId   string `gorm:"size:64;index:idx_video_time,priority:1;comment:video file id" json:"videoId,omitempty"`
	VideoTime *int64 `gorm:"index:idx_video_time,priority:2;comment:video progress(ms)" json:"videoTime,omitempty"`
	Status    int8   `gorm:"default:1;comment:1=normal 2=recall" json:"status"`
	SendTime  int64  `gorm:"not null;index:idx_send_time" json:"sendTime"`
}

File: frontend/src/db/models/message.js
export const MESSAGE_STORE_NAME = "messages";

export const MESSAGE_KEY_PATH = [
    "conversationId",
    "seq",
];

export const MESSAGE_INDEXES = {
    CONVERSATION: "conversation",
    CONVERSATION_SEQ: "conversationSeq",
};

/** 将消息转换为 IndexedDB 记录，只保留消息本体和业务状态。 */
export const createMessageRecord = (message) => {
    const seq = Number(message?.seq);
    // 正常消息使用 conversationId + seq 作为 IndexedDB 主键。
    if (!message?.conversationId || !message?.messageId || !Number.isSafeInteger(seq) || seq <= 0) {
        return null;
    }

    return {
        messageId: message.messageId,
        conversationId: message.conversationId,
        senderId: message.senderId || "",
        clientMsgId: message.clientMsgId || "",
        seq,
        cType: Number(message.cType) || 1,
        content: message.content || "",

        // // 弹幕相关接口兼容
        // videoId: message.videoId || "",
        // videoTime: message.videoTime ?? null,
        // 后端消息业务状态：1=normal，2=recall。
        status: Number(message.status) || 1,
        sendTime: Number(message.sendTime) || 0,
        attachmentId: message.attachmentId || "",
    };
};
